modules/SerialCommHub/tiny_modbus_rtu.cpp

```diff
@@ -64,117 +64,16 @@ static void append_checksum(uint8_t* msg, int msg_len) {
-        clear_exception_bit(function_code_recvd);
-    }
-
-    if (function != function_code_recvd) {
-        EVLOG_error << fmt::format("Function code mismatch: expected: {} received: {}", function, function_code_recvd);
-        return result;
-    }
-
-    if (!validate_checksum(buf, len)) {
-        EVLOG_error << "Checksum error";
-        return result;
-    }
 
-    if (!exception) {
-        // For a write reply we always get 4 bytes
-        uint8_t byte_cnt = 4;
-        int start_of_result = RES_TX_START_OF_PAYLOAD;
-
-        // Was it a read reply?
-        if (function == FunctionCode::READ_COILS || function == FunctionCode::READ_DISCRETE_INPUTS ||
-            function == FunctionCode::READ_MULTIPLE_HOLDING_REGISTERS ||
-            function == FunctionCode::READ_INPUT_REGISTERS) {
-            // adapt byte count and starting pos
-            byte_cnt = buf[RES_RX_LEN_POS];
-            start_of_result = RES_RX_START_OF_PAYLOAD;
-        }
-
-        // check if result is completely in received data
-        if (start_of_result + byte_cnt > len) {
-            EVLOG_error << "Result data not completely in received message.";
-            return result;
-        }
-
-        // ready to copy actual result data to output
-        result.reserve(byte_cnt / 2);
-
-        for (int i = start_of_result; i < start_of_result + byte_cnt; i += 2) {
-            uint16_t t;
-            memcpy(&t, buf + i, 2);
-            t = be16toh(t);
-            result.push_back(t);
-        }
-        return result;
-    } else {
-        // handle exception message
-        uint8_t err_code = buf[RES_RX_START_OF_PAYLOAD];
-        switch (err_code) {
-        case 0x01:
-            EVLOG_error << "Modbus exception: Illegal function";
-            break;
-        case 0x02:
-            EVLOG_error << "Modbus exception: Illegal data address";
-            break;
-        case 0x03:
-            EVLOG_error << "Modbus exception: Illegal data value";
-            break;
-        case 0x04:
-            EVLOG_error << "Modbus exception: Client device failure";
-            break;
-        case 0x05:
-            EVLOG_debug << "Modbus ACK";
-            break;
-        case 0x06:
-            EVLOG_error << "Modbus exception: Client device busy";
-            break;
-        case 0x07:
-            EVLOG_error << "Modbus exception: NACK";
-            break;
-        case 0x08:
-            EVLOG_error << "Modbus exception: Memory parity error";
-            break;
-        case 0x09:
-            EVLOG_error << "Modbus exception: Out of resources";
-            break;
-        case 0x0A:
-            EVLOG_error << "Modbus exception: Gateway path unavailable";
-            break;
-        case 0x0B:
-            EVLOG_error << "Modbus exception: Gateway target device failed to respond";
-            break;
-        }
-        return result;
-    }
+    EVLOG_info << "crc_sum: " << std::setfill('0') << std::setw(4) << std::hex << crc_sum;
```

Comment:
left over debug?

Reply:
Please make sure you comment only on the latest version, not on intermediary commits! 
This has already been fixed by the next commit.